pkg/trace/config/client.go

```diff
@@ -0,0 +1,69 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package config
+
+import (
+	"net/http"
+	"sync"
+	"time"
+)
+
+// TODO(gbbr): Perhaps this is not the best place for this structure.
+
+// ResetClient wraps (http.Client).Do and resets the underlying connections at the
+// configured interval
+type ResetClient struct {
```

Comment:
Is this copied over from another part of the codebase? If so, should we add a note about it?